### DISCUSSION TITLE
M4 option disappearing between runs

### DISCUSSION THREAD

**Discussion by MarkDowes** [arkypita/LaserGRBL#1182] [Q&A]
Hi I am having an issue where the M4 option is vanishing from existence between runs.

I launch LaserGRBL, I set up a project, i choose M4, i run it, no problems, then I go to run it again, and it has switched to M3 and M4 has disappeared altogether. I am changing nothing, not even relaunching the app, just reloading last file and M4 is not available when it was moments ago.

Sometimes resetting everything and relaunching the app will bring M4 back, sometimes it does not reappear and I fiddle around till it magically reappears again.
This is happening both with loading vectors for cutting, and with loading images for engraving.

Have wasted some money and materials in the process from over-burning because it has switched to M3 and I have not realised, unfortunately. 

I am actually stuck right now in that I cannot make M4 appear at all when I was using it 20 minutes ago. Nothing I am doing will make it reappear, and no grbl or LaserGRBL settings changed to make it vanish in the first place.

**Comment by arkypita** [comment]
http://lasergrbl.com/faq/

**Comment by arkypita** [comment]
You say you don't change anything. Maybe some custom button you have is changing $32 laser mode option.

**Comment by MarkDowes** [comment]
The only custom button I am using is the "turn on laser for focusing", none of the others. I see in editing that this changes S32 to 1 and 0, its also firing in M3 mode, would this effect it?

If so how is it that turning everything off, unplugging, restarting the computer, relaunching the app, does not make it reappear sometimes, but the other times it does?

**Comment by arkypita** [comment]
> I see in editing that this changes S32 to 1 and 0, its also firing in M3 mode, would this effect it?

Yes. 

I suggest you change like this:

![image](https://user-images.githubusercontent.com/8782035/100653106-32549d00-3348-11eb-81f6-bf7f8efa6743.png)

**Comment by arkypita** [comment]
> If so how is it that turning everything off, unplugging, restarting the computer, relaunching the app, does not make it reappear sometimes, but the other times it does?

You can turn on off PC, app, unplug etc but $ parameters are stored permanently inside the control board so they don't change.
LaserGRBL read them at first connection and when you open the edit grbl configuration dialog.

**Comment by MarkDowes** [comment]
Aaah ok, thank you very much for your help.

As always thank you for your continued development.
